Tasks/PerformActionOnMachineGroup/DtlHelper.ps1

```diff
@@ -10,10 +10,24 @@ function Get-MachineGroup
 {
     param([string]$machineGroupName,
           [string]$filters)
- 
-    Write-Verbose "Getting the machine group $machineGroupName" -Verbose
-    $environment = Get-Environment -EnvironmentName $machineGroupName  -Connection $connection -Filters $filters -ErrorAction Stop -Verbose
-    Write-Verbose "Retrieved the machine group"
+
+    if ($Action -eq "Block")
+    {
+        $getEnvironmentCommand = 
+        {
+            $environment = Get-Environment -EnvironmentName $machineGroupName  -Connection $connection -Filters $filters -ErrorAction Stop -Verbose
+        }
+        
+        Write-Verbose "Getting the machine group $machineGroupName" -Verbose
+        Invoke-WithRetry -Command $getEnvironmentCommand -RetryDuration $WaitTimeInMinutes -OperationDetail "Get Environment"
```

Comment:
you will need to add validations before using this value - as

$time = $waitTimeInMinutes -as [INT] 
96     if(($time -eq $null) -or ($time -lt 0)) 
97     { 
98         Write-Error("Cannot block machine group for $waitTimeInMinutes. Wait Time In Minutes should be a positive number which is minutes ...") 
99     }